bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java

```diff
@@ -605,8 +619,11 @@ private long addEventDescription(long dataSourceObjId, long fileObjId, Long arti
 			return Optional.empty();
 		}
 		long time = eventPayload.getTime();
-		// if the time is legitimate ( greater than zero ) insert it into the db
-		if (time <= 0) {
+		// if the time is legitimate ( greater than zero and less then 12 years from present time) insert it into the db
+		if (time <= 0 || time >= MAX_TIMESTAMP_TO_ADD) {
+			if (time >= MAX_TIMESTAMP_TO_ADD) {
+				logger.log(Level.WARNING, String.format("Date/Time discarded from Timeline for %s for artifact %s", artifact.getDisplayName(), eventPayload.getDescription(TimelineLevelOfDetail.HIGH)));
```

Comment:
Include the source file object id in the log.

Reply:
Fixed